lib/std/special/compiler_rt/atomics.zig

```diff
@@ -38,18 +45,35 @@ const SpinlockTable = struct {
     const Spinlock = struct {
         // Prevent false sharing by providing enough padding between two
         // consecutive spinlock elements
-        v: enum(usize) { Unlocked = 0, Locked } align(cache_line_size) = .Unlocked,
+        v: if (arch.isSPARC()) enum(u8) { Unlocked = 0, Locked = 255 } else enum(usize) { Unlocked = 0, Locked } align(cache_line_size) = .Unlocked,
 
         fn acquire(self: *@This()) void {
             while (true) {
-                switch (@atomicRmw(@TypeOf(self.v), &self.v, .Xchg, .Locked, .Acquire)) {
+                const flag = if (comptime arch.isSPARC())
+                    asm volatile ("ldstub [%[addr]], %[flag]"
+                        : [flag] "=r" (-> @TypeOf(self.v)),
+                        : [addr] "r" (&self.v),
+                        : "memory"
+                    )
+                else
```

Comment:
Nit: add brackets as done for `release` or remove them here.

Reply:
I tried to put brackets here but the compiler seems to not like it if there's brackets in if-expressions, that's why I left it out:

```
/home/k/zig/src/zig/lib/std/special/compiler_rt/atomics.zig:53:21: error: expression value is ignored
                    asm volatile ("ldstub [%[addr]], %[flag]"
                    ^
/home/k/zig/src/zig/lib/std/special/compiler_rt/atomics.zig:62:17: error: else prong required when switching on type 'void'
                switch (flag) {
                ^
```

Reply:
You need to use a labeled block and a break:
```zig
const val = val: {
    break :val 1;
};
```

Reply:
I see, thanks.